lib/Recur/RRuleIterator.php

```diff
@@ -588,6 +588,11 @@ protected function nextYearly() {
                     (int)$currentDayOfMonth
                 );
 
+                // maximum year 10000 should be safe to stop (2016 + max 3500 yearly occurrences)
+                if ((int)$currentYear > 10000) {
+                    throw new NoInstancesException('This recurrence rule does not generate any valid instances, stopped in year 10000');
```

Comment:
You might be right... But how do we leave the loop here without returning a "$this->currentDate"? Do we have to set it to NULL and then return?